ChangeLog.md

```diff
@@ -20,6 +20,13 @@ See docs/process.md for more on how version tagging works.
 
 2.0.21
 ------
+- Default shadow stack size used by emscripten reduced from 5Mb to 1Mb.  Anyone
+  running into issues can restore the old size using `-sTOTAL_STACK=5Mb`.  Since
+  windows has max stack of 1Mb and since WebAssembly stack usage should be less
+  than on native platforms (since its only used for address taken values) it
+  seems like 1Mb might still be on the high side here.  For reference, llvm
+  (`wasm-ld`) use 64kb as the default stack size.  `DEFAULT_PTHREAD_STACK_SIZE`
+  was also reduces from 2Mb to 1Mb to match primary stack.
```

Comment:
Would it be even better to match LLVM? It seems there is no reason to have differences in the wasm ecosystem here.

That is, maybe we should go to 64K here? Or maybe LLVM should go to 1MB? (I don't think there's any reason for 5MB, I agree with the logic there - I think it was originally because that's the linux stack size.)

Reply:
How about matching it to musl by reducing it to 128K?
https://wiki.musl-libc.org/functional-differences-from-glibc.html#Thread-stack-size
http://git.musl-libc.org/cgit/musl/commit/?id=c0058ab465e950c2c3302d2b62e21cc0b494224b.

Reply:
I think the linux stack is 8Mb.. so I'm not sure where 5Mb came from.. maybe asm.js needed a lot more of it?

Reply:
I agree it would make sense to have emscripten and llvm agree on the default.. but I fear that we could break more folks if we go down to 64k.     I'm trying to remember now why we chose 1 page (64K) in wasm-ld..

Reply:
Hmm, good question, yeah... Linux is 8. So not sure where 5 came from. Maybe a typo :wink:

Reply:
Looks like the default of 64Kb in wasm-ld goes way back to the initial commit: https://reviews.llvm.org/D34851

@sunfishcode would you object to doubling the default stack stack in the linker to 128k to match musl's default for new thread.   The goal would be to allow emscripten to use the same default as wasm-ld.